app/src/test/java/org/mozilla/fenix/ext/BookmarkNodeTest.kt

```diff
@@ -0,0 +1,84 @@
+@Config(application = TestApplication::class)
+class BookmarkNodeTest {
+    val bookMarkChild1 = newBookmarkNode("Child 1", 1, null)
+    val bookMarkChild2 = newBookmarkNode("Child 2", 2, null)
+    val allChildren = listOf(bookMarkChild1, bookMarkChild2)
+    val setofAllChildren = setOf(bookMarkChild1, bookMarkChild2)
+    val subsetToSubtract = setOf(bookMarkChild1)
+    val expectedRemainingSubset = listOf(bookMarkChild2)
+    val bookmarkNode = newBookmarkNode("Parent 1", 0, allChildren)
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing a sub set of children THEN the children subset is removed and rest remains`() {
+        val bookmarkNodeSubsetRemoved = bookmarkNode.minus(subsetToSubtract)
+        assertEquals(expectedRemainingSubset, bookmarkNodeSubsetRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing a set of all children THEN all children are removed`() {
+        val bookmarkNodeAllChildrenRemoved = bookmarkNode.minus(setofAllChildren)
+        assertEquals(emptyList<BookmarkNode>(), bookmarkNodeAllChildrenRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing a set of non-children THEN no children are removed`() {
+        val bookMarkChild3 = newBookmarkNode("Child 3", 3, null)
+        val setofNonChildren = setOf(bookMarkChild3)
+        val bookmarkNodeNonChildrenRemoved = bookmarkNode.minus(setofNonChildren)
+        assertEquals(allChildren, bookmarkNodeNonChildrenRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing an empty set THEN no children are removed`() {
+        val bookmarkNodeEmptySetRemoved = bookmarkNode.minus(emptySet())
+        assertEquals(allChildren, bookmarkNodeEmptySetRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with an empty list as children WHEN minusing a set of non-children from an empty parent THEN an empty list remains`() {
+        val parentWithEmptyList = newBookmarkNode("Parent 1", 0, emptyList<BookmarkNode>())
+        val parentWithEmptyListNonChildRemoved = parentWithEmptyList.minus(setofAllChildren)
+        assertEquals(emptyList<BookmarkNode>(), parentWithEmptyListNonChildRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with null as children WHEN minusing a set of non-children from a parent with null children THEN null remains`() {
+        val parentWithNullList = newBookmarkNode("Parent 1", 0, null)
+        val parentWithNullListNonChildRemoved = parentWithNullList.minus(setofAllChildren)
+        assertEquals(null, parentWithNullListNonChildRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing a sub-set of children THEN the rest of the parents object should remain the same`() {
+        val resultBookmarkNode = bookmarkNode.minus(subsetToSubtract)
+        val restOfResult = resultBookmarkNode.copy(children = expectedRemainingSubset)
```

Comment:
nit: I don't think it'd be immediately obvious what's going on here so I think this could be clearer with a comment that explains we're pinning `children` to the same value so we can compare the rest.